lib/connection.js

```diff
@@ -377,6 +397,8 @@ class Connection extends EventEmitter {
       this, application, authStrategy, credentials,
       this._onSessionCreated.bind(this)
     );
+
+    this.emit('handshakeRequest', applicationName, authStrategy, credentials);
```

Comment:
@belochub And I still see no harm in doing that. Let's wait what others have to say on this matter.